quill-core/src/main/scala/io/getquill/quotation/Quotation.scala

```diff
@@ -54,4 +74,32 @@ trait Quotation extends Parsing with Liftables with Unliftables {
       annotation <- method.annotations.headOption
       astTree <- annotation.tree.children.lastOption
     } yield (astTree)
+
+  private class FreeVarTraverser extends Traverser {
```

Comment:
I don't consider it a priority though, we could create a separate ticket to refactor it later.